#!/usr/bin/env python
PACKAGE = "base_vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_area", 				int_t,    0, "An Integer parameter", 200,	0,5000)
#blue, mode 0
gen.add("blue_H_low", 			int_t,    0, "An Integer parameter", 0,		0,179)
gen.add("blue_H_high", 			int_t,    0, "An Integer parameter", 179,	0,179)
gen.add("blue_S_low",  			int_t,    0, "An Integer parameter", 0,		0,255)
gen.add("blue_S_high", 			int_t,    0, "An Integer parameter", 65,	0,255)
gen.add("blue_V_low",  			int_t,    0, "An Integer parameter", 230,	0,255)
gen.add("blue_V_high", 			int_t,    0, "An Integer parameter", 255,	0,255)
#red: note that the LEDs light are very bright, thus we choose the V value 255 only, mode 0
gen.add("red_H_low1", 			int_t,    0, "An Integer parameter", 0,		0,179)
gen.add("red_H_high1", 			int_t,    0, "An Integer parameter", 90,	0,179)
gen.add("red_H_low2", 			int_t,    0, "An Integer parameter", 90,	0,179)
gen.add("red_H_high2", 			int_t,    0, "An Integer parameter", 179,	0,179)
gen.add("red_S_low",  			int_t,    0, "An Integer parameter", 0,		0,255)
gen.add("red_S_high", 			int_t,    0, "An Integer parameter", 65,	0,255)
gen.add("red_V_low",  			int_t,    0, "An Integer parameter", 240,	0,255)
gen.add("red_V_high", 			int_t,    0, "An Integer parameter", 255,	0,255)

# black, red armor: for the gray numbered circle, mode 0 & 1
gen.add("black_H_low_r", 		int_t,    0, "An Integer parameter", 0,		0,179)
gen.add("black_H_high_r", 	int_t,    0, "An Integer parameter", 179,	0,179)
gen.add("black_S_low_r",  	int_t,    0, "An Integer parameter", 0,		0,255)
gen.add("black_S_high_r", 	int_t,    0, "An Integer parameter", 255,	0,255)
gen.add("black_V_low_r", 		int_t,    0, "An Integer parameter", 0,		0,255)
gen.add("black_V_high_r",		int_t,    0, "An Integer parameter", 180,	0,255)
# black, blue armor: for the gray numbered circle, mode 0 & 1
gen.add("black_H_low_b", 		int_t,    0, "An Integer parameter", 0,		0,179)
gen.add("black_H_high_b", 	int_t,    0, "An Integer parameter", 179,	0,179)
gen.add("black_S_low_b", 		int_t,    0, "An Integer parameter", 80,	0,255)
gen.add("black_S_high_b",		int_t,    0, "An Integer parameter", 255,	0,255)
gen.add("black_V_low_b", 		int_t,    0, "An Integer parameter", 0,		0,255)
gen.add("black_V_high_b",		int_t,    0, "An Integer parameter", 254,	0,255)
#black: thresholded from gray frame in mode 2
gen.add("black_low_mode1", 	int_t,  	0, "An Integer parameter", 0,		0,255)
gen.add("black_high_mode1",	int_t,    0, "An Integer parameter", 130,	0,255)
#white: thresholded from gray frame in mode 2
gen.add("white_low_mode1", 	int_t,  	0, "An Integer parameter", 170,	0,255)
gen.add("white_high_mode1",	int_t,    0, "An Integer parameter", 255,	0,255)
# green, for 3x3 matrix detection
gen.add("green_H_low", 			int_t,    0, "An Integer parameter", 40,	0,179)
gen.add("green_H_high", 		int_t,    0, "An Integer parameter", 80,	0,179)
gen.add("green_S_low",  		int_t,    0, "An Integer parameter", 50,	0,255)
gen.add("green_S_high", 		int_t,    0, "An Integer parameter", 255,	0,255)
gen.add("green_V_low", 			int_t,    0, "An Integer parameter", 50,	0,255)
gen.add("green_V_high",			int_t,    0, "An Integer parameter", 255,	0,255)


exit(gen.generate(PACKAGE, "base_vision", "armor_color"))